plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/resource/ResourcePayloadPage.java

```diff
@@ -0,0 +1,284 @@
+                    String file = VariablesHelper.addWorkspacePrefix(path);
+                    setLocalSourceFileName(file);
+                }
+            }
+
+
+        };
+    }
+    
+    private void setLocalSourceFileName(String file) {
+        if (file == null || !isFile(file)) {
+            return;
+        }
+        try {
+            model.setLocalSourceFileName(file);
+        } catch (ClassCastException | OpenShiftException ex) {
+            IStatus status = ValidationStatus.error(ex.getMessage(), ex);
+            OpenShiftUIActivator.getDefault().getLogger().logStatus(status);
+            ErrorDialog.openError(getShell(), "Openshift resource error",
+                    NLS.bind("The file \"{0}\" is not an OpenShift resource.", 
+                            file),
+                    status);
+        }
+    }
+   
+    private void createRemoteSourceControls(DataBindingContext dbc, Group sourceGroup) {
+        Composite composite = new Composite(sourceGroup, SWT.NONE);
+        GridLayoutFactory.fillDefaults().numColumns(2).applyTo(composite);
+        GridDataFactory.fillDefaults().span(3, 1).grab(true,  false).applyTo(composite);
+        Button remoteButton = new Button(composite, SWT.RADIO);
+        final IObservableValue remoteTypeSelection = WidgetProperties.selection().observe(remoteButton);
+        ValueBindingBuilder
+        .bind(remoteTypeSelection)
+        .converting(new Boolean2EnumConverter(SourceType.REMOTE))
+        .to(BeanProperties.value(IResourcePayloadPageModel.PROPERTY_SOURCE_TYPE).observe(model))
+        .converting(new Enum2BooleanConverter(SourceType.REMOTE))
+        .in(dbc);
+
+        Label label = new Label(composite, SWT.NONE);
+        label.setText("Create from url");
+        
+        // local template file name
+        Text remoteSourceURLText = new Text(sourceGroup, SWT.BORDER);
+        GridDataFactory.fillDefaults()
+                .align(SWT.FILL, SWT.CENTER)
+                .grab(true, false)
+                .span(3, 1)
+                .applyTo(remoteSourceURLText);
+        final IObservableValue remoteSourceURL = WidgetProperties.text(SWT.Modify).observe(remoteSourceURLText);
+        ValueBindingBuilder
+                .bind(remoteSourceURL )
+                //.validatingBeforeSet(new SimpleUrlStringValidator())
+                .to(BeanProperties.value(
+                        IResourcePayloadPageModel.PROPERTY_REMOTE_SOURCE_URL).observe(model))
+                .in(dbc);
+        MultiValidator validator = new MultiValidator() {
+            
+            @Override
+            protected IStatus validate() {
+                String url = (String) remoteSourceURL.getValue();
+                return (boolean) remoteTypeSelection.getValue() && !UrlUtils.isValid(url)?ValidationStatus.error(NLS.bind(CommonUIMessages.URLSTRINGVALIDATOR_NOT_A_VALID_URL, url)):ValidationStatus.ok();
```

Comment:
Below URL_VALIDATOR.validate() is used, here UrlUtils.isValid(url). Can we do it as 

return (boolean) remoteTypeSelection.getValue() ? URL_VALIDATOR.validate(url) : ValidationStatus.ok();